### PR TITLE
Improve error messages with TVM_LOG_DEBUG and add docs

### PR DESCRIPTION
**Pull request by areusch**
This PR makes it easier to use TVM_LOG_DEBUG and documents it underneath Developer How-To.

@mbs-octoml

### CODE REVIEW

File: src/runtime/logging.cc [hunk 1]
```diff
@@ -222,33 +222,45 @@ TvmLogDebugSettings TvmLogDebugSettings::ParseSpec(const char* opt_spec) {
     return settings;
   }
   std::istringstream spec_stream(spec);
+  auto tell_pos = [&](const std::string& last_read) {
```
Comment (mbs-octoml) [inline src/runtime/logging.cc:225]: While you're in there could you remove the '/' prefix from kSrcPrefix (and change the length). I keep getting caught by src/x/y/z.cc=1 not working.

**Review by mbs-octoml — APPROVED** [review]
Hey thanks a ton!

File: src/runtime/logging.cc (hunk 1 above)
Comment (areusch) [reply src/runtime/logging.cc:231]: fixed

**Comment by areusch** [comment]
@leandron @Mousius can you approve?

**Review by masahi — APPROVED** [review]

**Pull request merged**